### ISSUE TITLE
[Plant Management] Research Per-plant Care Checklist

### ISSUE THREAD

**Issue by danielkhodge** [SEproject2025/greenhouse#72]
Every plant should have a set of checkboxes that are tasks associated with the plant's care.

Every plant has a different schedule of tasks that may appear every day.

When a new day arrives, the plants care queue (or the tasks that appear on the plant's tile) should change to only those tasks that are necessary to complete for that day.

When all tasks for the plant have been completed on the day, the plant's tile should noticeably change indicating that the plant has been successfully cared for.

Back-end: Each done task is marked for retrieval and once the day is done those tasks are submitted as complete.

Acceptance Criteria: Find out what is necessary to make this feature a reality. And if possible in the sprint,  complete the feature.

Source User Stories:
As a user I want to know my plant is safe and that I am done taking care of my plant for the day.
- Tile flips when all check boxes are checked 

As a user I want to know how to care for my plant 
- I see one or more check boxes in the plant tile; or possibly the text of the task gets crossed out when clicked on and completed 
- There could be some kind of info button for each card that tells how exactly to do each thing, and also lets the user see how often each care task actually needs to be done.

**Comment by prail** [comment]
To complete this issue, you will likely need to do a few things:

- [ ] Create a task entity with the ORM, speak to @AnnaMarie20 about doing this.
- [ ] Every task entity should be linked to a plant.
- [ ] Every plant entity/card should have a method on it that is called upon creation of the new day, that links the necessary task entities for the day to its record in the database.
- [ ] When a task entity is marked as complete, this should be reflected in the database.
- [ ] There should be some means for the UI to know if a plant has had all of its tasks completed for the day.
- [ ] There should be some means for the UI to know if a task was not scheduled for the day.
- [ ] Tasks are time bound, that is, the UI should not render a task if it is not a task for that day.

So the Task class/entity in the DB will need to have these attributes:
- Description
- Plant FK
- Task PK
- Date
- Completed state (boolean or enum)

A plant entity/class should have these additional attributes/methods
- means to render
- means to get all tasks for the plant
- means to create new task entities for the current day

**Comment by prail** [comment]
If you find that implementing this task, is simple enough and doesn't require a major re-write, you may begin implementation.
